os-observer/prisma/schema.prisma

```diff
@@ -7,61 +7,135 @@ datasource db {
+  // Human-readable description for the organization
+  description     String?
+
+  // RELATIONS
+  // Associated artifacts
+  artifacts       Artifact[]
+  // Associated events
+  events          Event[]
+  // Associated event source pointers
+  eventSourcePtrs EventSourcePointer[]
 }
 
-model wallets {
-  id         String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
-  created_at DateTime @default(now()) @map("created_at") @db.Timestamptz(3)
-  address    String?
-  project_id String   @db.Uuid
-  type       String?
-  ens        String?
+// A particular piece of work created by an organization
+// TODO: if we really want to have separate tables for wallets, repositories, etc, we can turn this into a union type later
+//    https://dev.to/reggi/working-with-unions-within-prisma-nm6
+model Artifact {
+  id              Int           @id @default(autoincrement())
+  createdAt       DateTime      @default(now()) @db.Timestamptz(3)
+  updatedAt       DateTime      @updatedAt
+
+  // Relation to the Organization
+  organization    Organization  @relation(fields: [organizationId], references: [id])
+  organizationId  Int
+  
+  // The type of artifact
+  type            ArtifactType
+  // The namespace of `name`
+  namespace       ArtifactNamespace
+  // A unique identifier for this artifact within the namespace
+  name            String
+  // An optional URL to learn more about this artifact
+  url             String?
+  // Catch-all for other data we want to store
+  details         Json?
 
-  project projects @relation(fields: [project_id], references: [id])
+  // RELATIONS
+  // Associated events
+  events          Event[]
+  // Associated event source pointers
+  eventSourcePtrs EventSourcePointer[]
 }
 
-model events {
-  id          String     @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
-  created_at  DateTime   @default(now()) @map("created_at") @db.Timestamptz(3)
-  project_id  String     @db.Uuid
-  event_time  DateTime?  @db.Timestamptz(3)
-  event_type  String?
-  contributor String?
-  amount      Float?
-  details     Json?
-  event_enum  EventType?
+// An event that happened to an artifact
+// An event stream is identified by its (organization, artifact, eventType)
+model Event {
+  id              Int           @id @default(autoincrement())
+  createdAt       DateTime      @default(now()) @db.Timestamptz(3)
+  updatedAt       DateTime      @updatedAt
 
-  project projects @relation(fields: [project_id], references: [id])
+  // Relation to the Organization
+  organization    Organization  @relation(fields: [organizationId], references: [id])
+  organizationId  Int
+  // Relation to the Artifact
+  artifact        Artifact      @relation(fields: [artifactId], references: [id])
+  artifactId      Int
+  // Type of event
+  eventType       EventType?
+  // Time the event happened
+  eventTime       DateTime?     @db.Timestamptz(3)
+  // Who is the source of the event?
+  contributor     String?
```

Comment:
I think we might want to move `contributor` to its own table sooner rather than later. A contributor can work on the project (eg, contribute code) and fund a project (eg, contribute money) and I want to be able to visualize those graphs.

Reply:
good idea! ill add that